aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -220,6 +223,11 @@ def get_install_action(
         BaseAction
             Install action to use
         """
+        if install_links and not NodejsNpmWorkflow.can_use_install_links(subprocess_npm):
+            raise OldNpmVersionError(
```

Comment:
Is there a way that we can fallback to not build in source behavior if this happens?

Reply:
Updated, now instead of hard failing, we log a warning and revert the build directory back to the workflow's default (artifacts for regular Node and scratch for esbuild).